src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -1527,5 +1598,360 @@ static void Main()
 
             var comp = CompileAndVerify(source, expectedOutput: "12345678901234567890123456789012345");
         }
+
+
+        [Fact]
+        public void TupleTargetTypeTwice()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        // this works
+        (short, string) x1 = (1, ""hello"");
+        // this does not
+        (short, string) x2 = ((byte, string))(1, ""hello"");
+    }
+}
+" + trivial2uple;
+
+            CreateCompilationWithMscorlib(source).VerifyDiagnostics(
+                // (9,30): error CS0029: Cannot implicitly convert type '<tuple: byte Item1, string Item2>' to '<tuple: short Item1, string Item2>'
+                //         (short, string) x2 = ((byte, string))(1, "hello");
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, @"((byte, string))(1, ""hello"")").WithArguments("<tuple: byte Item1, string Item2>", "<tuple: short Item1, string Item2>").WithLocation(9, 30)
+            );
+        }
+
+        [Fact]
+        public void TupleTargetTypeLambda()
+        {
+            var source = @"
+
+using System;
+
+class C
+{
+    static void Test(Func<Func<(short, short)>> d)
+    {
+        Console.WriteLine(""short"");
```

Comment:
This message will never get printed. Same thing for message for byte a couple lines below.